### ISSUE TITLE
Authentication - Google and Email Signin

### ISSUE THREAD

**Issue by OlukaDenis** [OlukaDenis/Justice-App#2]
Hello team,
This issue specifically deals with the following;

- Connect the app to google firebase (credentials will be provided)
- Authenticate google sign in
- Authenticate email sign in

This will also commerce ASAP, we shall just use the typical screen (common buttons or textviews) as we wait for the other group who is working on the login screens. The main task will be on the Java activities then connect to the firebase API.
I have created a branch named **google-email-signin** so you'll pull this branch to your local repo and then push all the progress there then later we shall delete it.

### Due date: 29th-August-2019

Hope to see progress soon

Denis
Android Project Lead.

**Comment by VICTORRELOADED** [comment]
What is the time frame

**Comment by OlukaDenis** [comment]
This should be done within two weeks

**Comment by joshmpaka** [comment]
Are we going to use the default firebase-ui or a custom sign in screen

**Comment by OlukaDenis** [comment]
We should use a custom one so that we have control over it. In case of any exception, we'll be able to trace it.
Thanks

**Comment by joshmpaka** [comment]
I believe all requirements for this issue have been met

**Comment by OlukaDenis** [comment]
I will have to merge changes for both branches together then we shall be able to mark these issues closed

**Comment by joshmpaka** [comment]
Ok, Boss

**Comment by OlukaDenis** [comment]
Hey @Josh-M-Dev , am in the process of merging the two branches but i have realised your code doesn't have the google-services.json file. It is missing.

**Comment by joshmpaka** [comment]
Yes.  i thought you would create a firebase  account and then just copy the google json file but if you need it i can upload it

**Comment by OlukaDenis** [comment]
Actually i had to create it:
justice2019.app@gmail.com
Password: Justice2019

We shall need those credentials throughout the development.
I went on to connect the app to the Firebase account

**Comment by joshmpaka** [comment]
Noted

**Comment by joshmpaka** [comment]
@OlukaDenis  I noticed you have enabled Phone verification. have you added that functionality or it was also required and hasnt been added yet.

**Comment by OlukaDenis** [comment]
Yeah it hasn't been added but we'll, lemme first get done with merging the two into one

**Comment by joshmpaka** [comment]
Ok

**Comment by joshmpaka** [comment]
@OlukaDenis Hello, i added the google json file for justice app to the project but the google sign in generated an error api exception 10. I did some digging and found out the most probable reason its not working is because of the SHA1 key provided(its a release key instead of debug key). Plus i also noticed the SHA1 key in your Travelmantics project is different from the one in the justice app. if you used the same computer for both projects then i think you might have provided a release key instead of debug key for the justice app.

**Comment by OlukaDenis** [comment]
You checkout the **denis** branch, i merged all the branches into one. It is working perfectly

**Comment by Bryanmuloni** [comment]
@OlukaDenis this is great, is it okay to pull from this branch? Besides, when is the next sprint starting we really do not have enough time remaining.
Thank you
